src/components/userModal/UserModal.js

```diff
@@ -0,0 +1,223 @@
+import React, { Component } from "react";
+import axios from 'axios';
+// import { Button } from "react-bootstrap";
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+// import Form from 'react-bootstrap/Form';
+import {locationIcon, infoIcon, emailIcon, phoneIcon, caseIcon, bookIcon, tickIcon, communityIcon, attachmentIcon} from "../../assets/index.js";
+import ImageFormatter from "../imageFormatter/ImageFormatter";
+import StatusFormatter from "../statusFormatter/StatusFormatter";
+import CareerInfo from "../careerInfo/CareerInfo"
+import * as Constants from "../../utils/constants";
+import PhoneNumberFormatter from "../phoneNumberFormatter/PhoneNumberFormatter"
+import {ToggleButton, ToggleButtonGroup, Button} from 'react-bootstrap';
+import './UserModal.scss'
+
+const ACCEPT = "Accept";
+const REJECT = "Reject";
+
+class UserModal extends Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            "status":"",
+            "rejectReason":""
+        }
+        this.changeStatus = this.changeStatus.bind(this);
+        this.setRejectReason = this.setRejectReason.bind(this);
+        this.submitStatusUdate = this.submitStatusUdate.bind(this);
+    }
+
+    changeStatus(status) {
+        this.setState({"status":status});
+        if(status === ACCEPT) {
+            this.setState({"status":Constants.userStatus.APPROVED})
+            this.setState({"rejectReason":""});
+        } else {
+            this.setState({"status":Constants.userStatus.REJECTED})
+        }
+    }
+
+    setRejectReason(e) {
+        this.setState({"rejectReason":e.target.value});
+    }
+
+    submitStatusUdate(e) {
+        e.preventDefault();
+
+        switch(this.state.status) {
+            case Constants.userStatus.APPROVED: 
+                console.log("Approved user_id:", this.props.data.user_id);
+                let copiedObject = JSON.parse(JSON.stringify(this.props.data))
+                copiedObject.status = this.state.status
+                axios.put(Constants.apiRootURL + '/users/' + this.props.data.user_id, copiedObject)
+                .then(response => response.data)
+                .then(result => this.props.onsubmitUpdate(result))
+                break;
+            case Constants.userStatus.REJECTED:
+                console.log("Deleting user_id:", this.props.data.user_id)
+                axios({
+                    method: 'DELETE',
+                    url: Constants.apiRootURL + '/users/' + this.props.data.user_id
+                })
+                .then(response => response.data)
+                .then(result => this.props.onsubmitUpdate(result));
+                break;
+            default:
+                alert("Please choose a valid option")
+        }
+    }
+
+    render() {
+        return (
```

Comment:
This component looks amazing, and great work on it! 
I think this is out of the scope of this PR, but in the future, we should make this component smaller and create separate components for different elements. As some of them our be reusable elsewhere too. If that is okay, I can create an issue on Zenhub for that and add that in backlog.

Reply:
I tried thinking about where else could it be used and how do I divide those. I could not think in terms of components and thus kept everything in one file. You could surely make an issue and keep it in the backlog

Reply:
Sure, created https://app.zenhub.com/workspaces/jjcsa-5ed1bcd022309703adc78079/issues/jjcsa/frontend/41
Feel free to assign to yourself, if you wish!